This patchset is for bug:

23033470 - oslo.db needs a database option to support MySQL Cluster

This fixes the following aspects of Oslo.db:
1. Implementation of an oslo.db configuration parameter to specify the MySQL 
   storage engine (mysql_storage_engine).

This has not been committed upstream, but has been filed in launchpad:

https://bugs.launchpad.net/oslo.db/+bug/1564110


--- oslo.db-1.7.2/oslo_db/options.py.orig	2016-02-17 14:56:11.276310637 -0700
+++ oslo.db-1.7.2/oslo_db/options.py	2016-02-19 10:32:20.551188376 -0700
@@ -50,6 +50,12 @@ database_opts = [
                     'server-set SQL mode. To use whatever SQL mode '
                     'is set by the server configuration, '
                     'set this to no value. Example: mysql_sql_mode='),
+    cfg.StrOpt('mysql_storage_engine',
+               default='InnoDB',
+               help='This configures the MySQL storage engine. '
+               'This allows for OpenStack to support different storage engines '
+               'such as InnoDB, NDB, etc. For MySQL Cluster, set to NDBCLUSTER. '
+               'Example: mysql_storage_engine=NDBCLUSTER'),
     cfg.IntOpt('idle_timeout',
                default=3600,
                deprecated_opts=[cfg.DeprecatedOpt('sql_idle_timeout',
